pyccel/ast/variable.py

```diff
@@ -863,15 +863,15 @@ class IndexedElement(TypedAstNode):
     True
     """
     __slots__ = ('_label', '_indices','_dtype','_precision','_shape','_rank','_order','_class_type')
-    _attribute_nodes = ('_label', '_indices')
+    _attribute_nodes = ('_label', '_indices', '_shape')
```

Comment:
yes otherwise it used the old `shape`, we probably need it also for `Variable`

Reply:
Then also for `DottedVariable` probably. We may also need it elsewhere :/

Reply:
isn't the `shape` of `DottedVariable` the same as the `shape` of the last variable in the `DottedVariable` ?

Reply:
A `DottedVariable` *is* a `Variable` it doesn't contain a list of `Variable`s, but you are right this is handled via inheritance

Reply:
I don't think there is an issue in `Variable`, because we substitute the whole variable, the same thing goes for `DottedVariable`, the issue is only present in the IndexedElement, because it contains a different `shape` then its base